core/includes/theme.inc

```diff
@@ -1466,7 +1466,14 @@ function theme_get_setting($setting_name, $theme = NULL) {
 
   // Loop up through any base themes.
   if (is_null($value) && isset($theme_info->info['base theme'])) {
-    $value = theme_get_setting($setting_name, $theme_info->info['base theme']);
+    $theme = $theme_info->info['base theme'];
+    $value = theme_get_setting($setting_name, $theme);
+  }
+
+  // Save the setting to config.
+  if (!is_null($value)) {
+    $configs[$theme]->set($setting_name, $value);
```

Comment:
This would copy a setting from the base-theme to the sub-theme, but it wouldn't change the behavior at all because the base-theme already provides this value if it's not in the sub-theme. 

If we do want the settings from a base-theme to be copied to the config file for the sub-theme then we should do them all at once when the sub-theme is enabled. It should't happen one setting at a time, and certainly not during a read.